src/components/SidebarComp.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div class="sidebar">
+    <router-link to="/home">
+      <div>
+        <HomeAccount />
+        <span>Main Menu</span>
+      </div>
+    </router-link>
+    <router-link to="/add">
+      <div>
+        <BookPlus />
+        <span>Add Books</span>
+      </div>
+    </router-link>
+    <router-link to="/issueBook">
+      <div>
+        <BookCheck />
+        <span>Issue Books</span>
+      </div>
+    </router-link>
+    <router-link to="/custDetails">
+      <div>
+        <AccountDetails />
+        <span>{{ sidebarTitle }}</span>
+      </div>
+    </router-link>
+  </div>
+</template>
+
+<script>
+import HomeAccount from "vue-material-design-icons/HomeAccount.vue";
+import BookPlus from "vue-material-design-icons/BookPlus.vue";
+import BookCheck from "vue-material-design-icons/BookCheck.vue";
+import AccountDetails from "vue-material-design-icons/AccountDetails.vue";
+
+export default {
+  name: "SidebarComp",
+  props: {},
+
+  data() {
+    return {
+      role: "",
+      userRole: "",
+    };
+  },
+  beforeMount() {
+    let user = JSON.parse(localStorage.getItem("user-info"));
+    this.userRole = user.role;
+    if (!user) {
```

Comment:
this also you can add in just one file, the main file where the app starts and don't use it in every file.